<a href="https://colab.research.google.com/github/duonghiepit/data_analyst/blob/main/Profitable_App_Profiles_for_Google_Play_and_App_Store_Markets.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Profitable App Profiles for the App Store and Google Play Markets**

Hồ sơ ứng dụng có lợi nhuận cho App Store và Google Play Markets

This project is aimed at finding the profitable mobile app profiles for the App Store and Google Play markets. The goal is to analyze data to help developers understand types of Android and iOS application that would attract more users. Suppose that the app is free to download and is meant for English-speaking users.

The project is my very first touch with Data Analytics utilizing Python. My goal with this project is acknowledge that even the very fundamental Python programming language can be used to analyze data. It took me 2 days working consistenty and continuously to finish this project. I am very glad that I have done my very first analytics project with Python and looking forward for upcoming projects in Data Analytics.

---

Dự án này nhằm mục đích tìm kiếm hồ sơ ứng dụng di động sinh lời cho thị trường App Store và Google Play. Mục tiêu là phân tích dữ liệu để giúp các nhà phát triển hiểu các loại ứng dụng Android và iOS sẽ thu hút nhiều người dùng hơn. Giả sử ứng dụng được tải xuống miễn phí và dành cho người dùng nói tiếng Anh.

Dự án này là lần đầu tiên tôi tiếp xúc với Phân tích dữ liệu bằng cách sử dụng Python. Mục tiêu của tôi với dự án này là thừa nhận rằng ngay cả ngôn ngữ lập trình Python rất cơ bản cũng có thể được sử dụng để phân tích dữ liệu. Tôi đã mất 2 ngày làm việc kiên trì và liên tục để hoàn thành dự án này. Tôi rất vui vì tôi đã thực hiện dự án phân tích đầu tiên của mình với Python và mong chờ các dự án sắp tới trong Phân tích dữ liệu.

## **Opening data**

---

According to statistics on statista, in the third quarter of 2019, there are 2.47 million apps on Google Play and 1.8 million apps on Apple's App Store. Hence, there would be around 4.27 million apps in total. Collecting data for over 4 million apps requires abundant amount of time and cost. Therefore, smaller data sets of apps would be utilized instead as for the scope of this project:

---

Theo thống kê trên Statista, trong quý 3 năm 2019, có 2,47 triệu ứng dụng trên Google Play và 1,8 triệu ứng dụng trên App Store của Apple. Do đó, tổng cộng sẽ có khoảng 4,27 triệu ứng dụng. Việc thu thập dữ liệu của hơn 4 triệu ứng dụng đòi hỏi nhiều thời gian và chi phí. Do đó, các bộ dữ liệu ứng dụng nhỏ hơn sẽ được sử dụng thay thế cho phạm vi của dự án này:



*   [Google Play market](https://www.kaggle.com/datasets/lava18/google-play-store-apps): around 10,000 apps.
*   [Apple's App Store](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps): around 7,000 apps



In [94]:
# Read in data
import numpy as np
import pandas as pd

andr_df = pd.read_csv('googleplaystore.csv')
ios_df = pd.read_csv('AppleStore.csv')

In [95]:
andr_df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [96]:
andr_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [97]:
ios_df.head()

Unnamed: 0         id                                         track_name  \
0           1  281656475                                    PAC-MAN Premium   
1           2  281796108                          Evernote - stay organized   
2           3  281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   
3           4  282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   
4           5  282935706                                              Bible   

   size_bytes currency  price  rating_count_tot  rating_count_ver  \
0   100788224      USD   3.99             21292                26   
1   158578688      USD   0.00            161065                26   
2   100524032      USD   0.00            188583              2822   
3   128512000      USD   0.00            262241               649   
4    92774400      USD   0.00            985920              5320   

   user_rating  user_rating_ver     ver cont_rating   prime_genre  \
0          4.0              4.5   6.3.5          4+         Games   
1          4.0              3.5   8.2.2          4+  Productivity   
2          3.5              4.5   5.0.0          4+       Weather   
3          4.0              4.5  5.10.0         12+      Shopping   
4          4.5              5.0   7.5.1          4+     Reference   

   sup_devices.num  ipadSc_urls.num  lang.num  vpp_lic  
0               38                5        10        1  
1               37                5        23        1  
2               37                5         3        1  
3               37                5         9        1  
4               37                5        45        1

In [98]:
ios_df.drop(columns=['Unnamed: 0', 'id'], inplace=True)

In [99]:
ios_df.columns

Index(['track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot',
       'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver',
       'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num',
       'lang.num', 'vpp_lic'],
      dtype='object')

In [100]:
andr_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [101]:
ios_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7197 entries, 0 to 7196
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   track_name        7197 non-null   object 
 1   size_bytes        7197 non-null   int64  
 2   currency          7197 non-null   object 
 3   price             7197 non-null   float64
 4   rating_count_tot  7197 non-null   int64  
 5   rating_count_ver  7197 non-null   int64  
 6   user_rating       7197 non-null   float64
 7   user_rating_ver   7197 non-null   float64
 8   ver               7197 non-null   object 
 9   cont_rating       7197 non-null   object 
 10  prime_genre       7197 non-null   object 
 11  sup_devices.num   7197 non-null   int64  
 12  ipadSc_urls.num   7197 non-null   int64  
 13  lang.num          7197 non-null   int64  
 14  vpp_lic           7197 non-null   int64  
dtypes: float64(3), int64(7), object(5)
memory usage: 843.5+ KB


In [102]:
andr_df.describe()

Rating
count  9367.000000
mean      4.193338
std       0.537431
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max      19.000000

In [103]:
ios_df.describe()

size_bytes        price  rating_count_tot  rating_count_ver  \
count  7.197000e+03  7197.000000      7.197000e+03       7197.000000   
mean   1.991345e+08     1.726218      1.289291e+04        460.373906   
std    3.592069e+08     5.833006      7.573941e+04       3920.455183   
min    5.898240e+05     0.000000      0.000000e+00          0.000000   
25%    4.692275e+07     0.000000      2.800000e+01          1.000000   
50%    9.715302e+07     0.000000      3.000000e+02         23.000000   
75%    1.819249e+08     1.990000      2.793000e+03        140.000000   
max    4.025970e+09   299.990000      2.974676e+06     177050.000000   

       user_rating  user_rating_ver  sup_devices.num  ipadSc_urls.num  \
count  7197.000000      7197.000000      7197.000000      7197.000000   
mean      3.526956         3.253578        37.361817         3.707100   
std       1.517948         1.809363         3.737715         1.986005   
min       0.000000         0.000000         9.000000         0.000000   
25%       3.500000         2.500000        37.000000         3.000000   
50%       4.000000         4.000000        37.000000         5.000000   
75%       4.500000         4.500000        38.000000         5.000000   
max       5.000000         5.000000        47.000000         5.000000   

          lang.num      vpp_lic  
count  7197.000000  7197.000000  
mean      5.434903     0.993053  
std       7.919593     0.083066  
min       0.000000     0.000000  
25%       1.000000     1.000000  
50%       1.000000     1.000000  
75%       8.000000     1.000000  
max      75.000000     1.000000

In [104]:
andr_df.shape

(10841, 13)

In [105]:
ios_df.shape

(7197, 15)

## Data cleaning

### Remove Duplicate Entries

In [110]:
# Duplicate samples

a = andr_df.duplicated(subset=['App'])
b = ios_df.duplicated(subset=['track_name'])
print(andr_df[a][['App']])
print(ios_df[b][['track_name']])

                                      App
229          Quick PDF Scanner + OCR FREE
236                                   Box
239                    Google My Business
256                   ZOOM Cloud Meetings
261             join.me - Simple Meetings
...                                   ...
10715                  FarmersOnly Dating
10720  Firefox Focus: The privacy browser
10730                         FP Notebook
10753      Slickdeals: Coupons & Shopping
10768                                AAFP

[1181 rows x 1 columns]
               track_name
5603    VR Roller Coaster
7128  Mannequin Challenge


In [111]:
andr_df = andr_df.drop_duplicates(subset=['App']).reset_index(drop=True)
ios_df = ios_df.drop_duplicates(subset=['track_name']).reset_index(drop=True)

In [112]:
andr_df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [113]:
ios_df.head()

track_name  size_bytes currency  \
0                                    PAC-MAN Premium   100788224      USD   
1                          Evernote - stay organized   158578688      USD   
2    WeatherBug - Local Weather, Radar, Maps, Alerts   100524032      USD   
3  eBay: Best App to Buy, Sell, Save! Online Shop...   128512000      USD   
4                                              Bible    92774400      USD   

   price  rating_count_tot  rating_count_ver  user_rating  user_rating_ver  \
0   3.99             21292                26          4.0              4.5   
1   0.00            161065                26          4.0              3.5   
2   0.00            188583              2822          3.5              4.5   
3   0.00            262241               649          4.0              4.5   
4   0.00            985920              5320          4.5              5.0   

      ver cont_rating   prime_genre  sup_devices.num  ipadSc_urls.num  \
0   6.3.5          4+         Games               38                5   
1   8.2.2          4+  Productivity               37                5   
2   5.0.0          4+       Weather               37                5   
3  5.10.0         12+      Shopping               37                5   
4   7.5.1          4+     Reference               37                5   

   lang.num  vpp_lic  
0        10        1  
1        23        1  
2         3        1  
3         9        1  
4        45        1

In [115]:
andr_df.shape, ios_df.shape

((9660, 13), (7195, 15))

### Remove Non-English apps

For the purpose of this project, all the applications that are not meant for English-speaking users will also be removed. In order to reduce data loss, only application with its name containing more than 3 non-ASCII will be removed.

---

Vì mục đích của dự án này, tất cả các ứng dụng không dành cho người dùng nói tiếng Anh cũng sẽ bị xóa. Để giảm mất dữ liệu, chỉ ứng dụng có tên chứa nhiều hơn 3 ký tự không phải ASCII mới bị xóa.

In [53]:
def has_non_ascii_chars(s):
    non_ascii_count = sum(1 for char in s if ord(char) > 127)
    return non_ascii_count > 3

In [54]:
has_non_ascii_chars('My God :)')

False

In [117]:
filtered_andr_df = andr_df[~andr_df['App'].apply(has_non_ascii_chars)].reset_index(drop=True)
filtered_ios_df = ios_df[~ios_df['track_name'].apply(has_non_ascii_chars)].reset_index(drop=True)

In [118]:
filtered_andr_df.shape, filtered_ios_df.shape

((9615, 13), (6181, 15))

Total: 9615 Android apps and 6181 iOS apps left.

Tổng cộng: Còn lại 9615 ứng dụng Android và 6181 ứng dụng IOS.

In [65]:
filtered_andr_df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [119]:
filtered_ios_df.head()

track_name  size_bytes currency  \
0                                    PAC-MAN Premium   100788224      USD   
1                          Evernote - stay organized   158578688      USD   
2    WeatherBug - Local Weather, Radar, Maps, Alerts   100524032      USD   
3  eBay: Best App to Buy, Sell, Save! Online Shop...   128512000      USD   
4                                              Bible    92774400      USD   

   price  rating_count_tot  rating_count_ver  user_rating  user_rating_ver  \
0   3.99             21292                26          4.0              4.5   
1   0.00            161065                26          4.0              3.5   
2   0.00            188583              2822          3.5              4.5   
3   0.00            262241               649          4.0              4.5   
4   0.00            985920              5320          4.5              5.0   

      ver cont_rating   prime_genre  sup_devices.num  ipadSc_urls.num  \
0   6.3.5          4+         Games               38                5   
1   8.2.2          4+  Productivity               37                5   
2   5.0.0          4+       Weather               37                5   
3  5.10.0         12+      Shopping               37                5   
4   7.5.1          4+     Reference               37                5   

   lang.num  vpp_lic  
0        10        1  
1        23        1  
2         3        1  
3         9        1  
4        45        1

### Isolating the Free Apps

In [132]:
free_appandr_df = filtered_andr_df.loc[filtered_andr_df['Price']=='0']
free_appios_df = filtered_ios_df.loc[filtered_ios_df['price']==0]

In [133]:
free_appandr_df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [134]:
free_appios_df.head()

track_name  size_bytes currency  \
1                          Evernote - stay organized   158578688      USD   
2    WeatherBug - Local Weather, Radar, Maps, Alerts   100524032      USD   
3  eBay: Best App to Buy, Sell, Save! Online Shop...   128512000      USD   
4                                              Bible    92774400      USD   
6             PayPal - Send and request money safely   227795968      USD   

   price  rating_count_tot  rating_count_ver  user_rating  user_rating_ver  \
1    0.0            161065                26          4.0              3.5   
2    0.0            188583              2822          3.5              4.5   
3    0.0            262241               649          4.0              4.5   
4    0.0            985920              5320          4.5              5.0   
6    0.0            119487               879          4.0              4.5   

      ver cont_rating   prime_genre  sup_devices.num  ipadSc_urls.num  \
1   8.2.2          4+  Productivity               37                5   
2   5.0.0          4+       Weather               37                5   
3  5.10.0         12+      Shopping               37                5   
4   7.5.1          4+     Reference               37                5   
6  6.12.0          4+       Finance               37                0   

   lang.num  vpp_lic  
1        23        1  
2         3        1  
3         9        1  
4        45        1  
6        19        1

In [135]:
free_appandr_df.shape, free_appios_df.shape

((8862, 13), (3220, 15))

There are 8862 apps in Google Play store and 3220 apps in App Store for final analysis

Có 8862 ứng dụng trong cửa hàng Google Play và 3220 ứng dụng trong App Store để phân tích cuối cùng

## Analyzing data

As mentioned from the beginning, the aim is to determine type of applications that are likely to attract mỏe users since revenue heavily depends on the amount of users using these applications. To minimize overhead and risks, the validation strategy for an app idea consists of 3 steps:

1.   Build a minimal Android version of the app, and add it to Google Play store.
2.   If the app has good responses from users, the app will be kept developing.
3.   If the app is profitable after 6 months, an iOS version of app will be built and add to App Store.

It is important that the app profile should be successful on both Google Play and App Store markets.

---

Như đã đề cập từ đầu, mục đích là xác định loại ứng dụng có khả năng thu hút người dùng nhiều hơn vì doanh thu phụ thuộc rất nhiều vào lượng người dùng sử dụng các ứng dụng này. Để giảm thiểu chi phí và rủi ro, chiến lược xác thực ý tưởng ứng dụng bao gồm 3 bước:


1.   Xây dựng phiên bản Android tối thiểu của ứng dụng và thêm nó vào cửa hàng Google Play.
2.   Nếu ứng dụng có phản hồi tốt từ người dùng, ứng dụng sẽ tiếp tục phát triển.
3.   Nếu ứng dụng có lãi sau 6 tháng, phiên bản ứng dụng iOS sẽ được xây dựng và thêm vào App Store.

Điều quan trọng là hồ sơ ứng dụng phải thành công trên cả thị trường Google Play và App Store.

### Most common apps by genre

In [142]:
free_appandr_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [152]:
genre_andr_counts = free_appandr_df['Genres'].value_counts()
genre_andr_percentages = (genre_andr_counts / len(free_appandr_df)) * 100
genre_andr_percentages

Tools                          8.429248
Entertainment                  6.070864
Education                      5.348680
Business                       4.592643
Lifestyle                      3.893026
                                 ...   
Strategy;Action & Adventure    0.011284
Art & Design;Pretend Play      0.011284
Arcade;Pretend Play            0.011284
Entertainment;Education        0.011284
Strategy;Creativity            0.011284
Name: Genres, Length: 114, dtype: float64

In [154]:
genre_ios_counts = free_appios_df['prime_genre'].value_counts()
genre_ios_percentages = (genre_ios_counts / len(free_appios_df)) * 100
genre_ios_percentages

Games                58.136646
Entertainment         7.888199
Photo & Video         4.968944
Education             3.664596
Social Networking     3.291925
Shopping              2.608696
Utilities             2.515528
Sports                2.142857
Music                 2.049689
Health & Fitness      2.018634
Productivity          1.739130
Lifestyle             1.583851
News                  1.335404
Travel                1.242236
Finance               1.118012
Weather               0.869565
Food & Drink          0.807453
Reference             0.559006
Business              0.527950
Book                  0.434783
Navigation            0.186335
Medical               0.186335
Catalogs              0.124224
Name: prime_genre, dtype: float64

### Most popular apps by genre on the App Store

In [159]:
rating_count_total_iosapp = free_appios_df.groupby('prime_genre').sum()['rating_count_tot']

<ipython-input-159-0e4d7df584fe>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  rating_count_total_iosapp = free_appios_df.groupby('prime_genre').sum()['rating_count_tot']


In [162]:
rating_count_total_iosapp

prime_genre
Book                   556619
Business               127349
Catalogs                16016
Education              826470
Entertainment         3563577
Finance               1132846
Food & Drink           866682
Games                42705795
Health & Fitness      1514371
Lifestyle              840774
Medical                  3672
Music                 3783551
Navigation             516542
News                   913665
Photo & Video         4550647
Productivity          1177591
Reference             1348958
Shopping              2261254
Social Networking     7584125
Sports                1587614
Travel                1129752
Utilities             1513441
Weather               1463837
Name: rating_count_tot, dtype: int64

In [170]:
avg_ratings_ios = rating_count_total_iosapp/genre_ios_counts

In [174]:
avg_ratings_ios.sort_values(ascending=False)

Navigation           86090.333333
Reference            74942.111111
Social Networking    71548.349057
Music                57326.530303
Weather              52279.892857
Book                 39758.500000
Food & Drink         33333.923077
Finance              31467.944444
Photo & Video        28441.543750
Travel               28243.800000
Shopping             26919.690476
Health & Fitness     23298.015385
Sports               23008.898551
Games                22812.924679
News                 21248.023256
Productivity         21028.410714
Utilities            18684.456790
Lifestyle            16485.764706
Entertainment        14029.830709
Business              7491.117647
Education             7003.983051
Catalogs              4004.000000
Medical                612.000000
dtype: float64

Navigation apps have the highest number of user reviews, following by Reference and Social Networking genres.

---

Ứng dụng điều hướng có số lượng đánh giá của người dùng cao nhất, tiếp theo là thể loại Tham khảo và Mạng xã hội.

In [181]:
free_appios_df.loc[free_appios_df['prime_genre']=='Navigation', ['track_name','rating_count_tot']].sort_values(by='rating_count_tot', ascending=False)

track_name  rating_count_tot
172    Waze - GPS Navigation, Maps & Real-time Traffic            345046
1523                Google Maps - Navigation & Transit            154911
198                                        Geocaching®             12811
1095       CoPilot GPS – Car Navigation & Offline Maps              3582
275   ImmobilienScout24: Real Estate Search in Germany               187
874                               Railway Route Search                 5

Navigation applications are heavily influenced by Waze and Google Maps, which have around half a million users's reviews. The same applies for social media applications with giants such as Facebook, Skype and Pinterest and very few smaller key players such as Spotify and Shazam heavily influencing the average number. Let's take a look at Reference genres:

---
Các ứng dụng điều hướng bị ảnh hưởng nặng nề bởi Waze và Google Maps, những ứng dụng có khoảng nửa triệu lượt đánh giá của người dùng. Điều tương tự cũng áp dụng cho các ứng dụng truyền thông xã hội với những gã khổng lồ như Facebook, Skype và Pinterest và rất ít những ứng dụng chủ chốt nhỏ hơn như Spotify và Shazam ảnh hưởng nặng nề đến con số trung bình. Chúng ta hãy xem các thể loại tham khảo:


In [182]:
free_appios_df.loc[free_appios_df['prime_genre']=='Reference', ['track_name','rating_count_tot']].sort_values(by='rating_count_tot', ascending=False)

track_name  rating_count_tot
4                                                 Bible            985920
114               Dictionary.com Dictionary & Thesaurus            200047
368      Dictionary.com Dictionary & Thesaurus for iPad             54175
635                                    Google Translate             26786
483   Muslim Pro: Ramadan 2017 Prayer Times, Azan, Q...             18418
5578  New Furniture Mods - Pocket Wiki & Game Tools ...             17588
554                          Merriam-Webster Dictionary             16849
935                                           Night Sky             12122
5670  City Maps for Minecraft PE - The Best Maps for...              8535
5613  LUCKY BLOCK MOD ™ for Minecraft PC Edition - T...              4693
5711    GUNS MODS for Minecraft PC Edition - Mods Tools              1497
5629  Guides for Pokémon GO - Pokemon GO News and Ch...               826
1740                                               WWDC               762
5630  Horror Maps for Minecraft PE - Download The Sc...               718
3496                                        VPN Express                14
5962    Real Bike Traffic Rider Virtual Reality Glasses                 8
1879  Jishokun-Japanese English Dictionary & Translator                 0
3270                                            教えて!goo                 0

Once again, Bible and Dictionary.com skew up the average rating. It seems that the average number of ratings seem to be skewed up by few apps that have hundreds of thousanfs of user rating, while the other apps may struggle to attract users. We could get a bigger picture by removing these giants for ech genre and then recount the averages, but this requires a higher level of details, which is not suitable for this level of coding.


---

Một lần nữa, Bible và Dictionary.com lại nâng cao xếp hạng trung bình. Có vẻ như số lượng xếp hạng trung bình dường như bị sai lệch bởi một số ứng dụng có hàng trăm nghìn xếp hạng của người dùng, trong khi các ứng dụng khác có thể gặp khó khăn trong việc thu hút người dùng. Chúng ta có thể có được bức tranh lớn hơn bằng cách loại bỏ những phần lớn này cho thể loại ech và sau đó tính lại mức trung bình, nhưng điều này đòi hỏi mức độ chi tiết cao hơn, không phù hợp với cấp độ mã hóa này.

Some other popular genres include: weather, book, food and drink, finance and music.


*   Weather apps: people do not really need to spend too much time in-app, so the chance of making profit in-app adds are pretty low. Additionally, getting reliable live weather data may require connection to non-free APIs.
*   Food & Drink apps: these applications require actual cooking and a delivery service, which is outside the scope of project.
*   Finance apps: Building these apps requires domain knowledge and a finance expert, which is quite extra to build an app.
*   Music: this requires the understanding of license and copyright, a bit too complicated.
*   Book: seems to be most relevant one. If we can combine somehow to have some sort of dictionary or make the app towards gamification, that would be the best.

---
Một số thể loại phổ biến khác bao gồm: thời tiết, sách, đồ ăn thức uống, tài chính và âm nhạc.


*   Ứng dụng thời tiết: mọi người không thực sự cần dành quá nhiều thời gian trong ứng dụng, vì vậy cơ hội kiếm được lợi nhuận từ việc bổ sung trong ứng dụng là khá thấp. Ngoài ra, để có được dữ liệu thời tiết trực tiếp đáng tin cậy có thể yêu cầu kết nối với các API không miễn phí.
*   Ứng dụng Thực phẩm & Đồ uống: các ứng dụng này yêu cầu dịch vụ nấu ăn thực tế và giao hàng, nằm ngoài phạm vi dự án.
*   Ứng dụng tài chính: Việc xây dựng các ứng dụng này đòi hỏi kiến ​​thức về miền và chuyên gia tài chính, điều này khá khó để xây dựng một ứng dụng.
*   Âm nhạc: cái này đòi hỏi sự hiểu biết về giấy phép và bản quyền, hơi phức tạp một chút.
*   Sách: có vẻ là cuốn sách phù hợp nhất. Nếu chúng ta có thể kết hợp bằng cách nào đó để có một số loại từ điển hoặc làm cho ứng dụng theo hướng trò chơi hóa thì đó sẽ là điều tốt nhất.





Let's move to Google Play data set and what are the outcomes then.

---

Hãy chuyển sang tập dữ liệu Google Play và kết quả sau đó là gì.

In [185]:
free_appandr_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [190]:
free_appandr_df.head(3)

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   

  Reviews  Size    Installs  Type Price Content Rating  \
0     159   19M     10,000+  Free     0       Everyone   
1     967   14M    500,000+  Free     0       Everyone   
2   87510  8.7M  5,000,000+  Free     0       Everyone   

                      Genres      Last Updated Current Ver   Android Ver  
0               Art & Design   January 7, 2018       1.0.0  4.0.3 and up  
1  Art & Design;Pretend Play  January 15, 2018       2.0.0  4.0.3 and up  
2               Art & Design    August 1, 2018       1.2.4  4.0.3 and up

In [194]:
def cvt_installs(series):
    return float(series.replace(',','').strip('+'))

In [195]:
free_appandr_df['Installs'] = free_appandr_df['Installs'].apply(cvt_installs)

<ipython-input-195-73d5ebc88431>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  free_appandr_df['Installs'] = free_appandr_df['Installs'].apply(cvt_installs)


In [196]:
free_appandr_df.head(3)

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   

  Reviews  Size   Installs  Type Price Content Rating  \
0     159   19M    10000.0  Free     0       Everyone   
1     967   14M   500000.0  Free     0       Everyone   
2   87510  8.7M  5000000.0  Free     0       Everyone   

                      Genres      Last Updated Current Ver   Android Ver  
0               Art & Design   January 7, 2018       1.0.0  4.0.3 and up  
1  Art & Design;Pretend Play  January 15, 2018       2.0.0  4.0.3 and up  
2               Art & Design    August 1, 2018       1.2.4  4.0.3 and up

In [207]:
rating_count_total_andrapp = free_appandr_df.groupby('Category')['Installs'].sum()

In [210]:
rating_count_total_andrapp

Category
ART_AND_DESIGN         1.143211e+08
AUTO_AND_VEHICLES      5.308006e+07
BEAUTY                 2.719705e+07
BOOKS_AND_REFERENCE    1.665884e+09
BUSINESS               6.969021e+08
COMICS                 4.497115e+07
COMMUNICATION          1.103691e+10
DATING                 1.409148e+08
EDUCATION              3.513500e+08
ENTERTAINMENT          2.113460e+09
EVENTS                 1.597316e+07
FAMILY                 4.400796e+09
FINANCE                4.551631e+08
FOOD_AND_DRINK         2.117388e+08
GAME                   1.385787e+10
HEALTH_AND_FITNESS     1.143548e+09
HOUSE_AND_HOME         9.721246e+07
LIBRARIES_AND_DEMO     5.299581e+07
LIFESTYLE              4.974844e+08
MAPS_AND_NAVIGATION    5.030608e+08
MEDICAL                3.763234e+07
NEWS_AND_MAGAZINES     2.368196e+09
PARENTING              3.147101e+07
PERSONALIZATION        1.529236e+09
PHOTOGRAPHY            4.647269e+09
PRODUCTIVITY           5.791629e+09
SHOPPING               1.400339e+09
SOCIAL             

In [211]:
category_andr_counts = free_appandr_df['Category'].value_counts()

In [214]:
avg_n_installs = rating_count_total_andrapp / category_andr_counts
avg_n_installs.sort_values(ascending=False)

COMMUNICATION          3.845612e+07
VIDEO_PLAYERS          2.485273e+07
SOCIAL                 2.325365e+07
ENTERTAINMENT          2.113460e+07
PHOTOGRAPHY            1.780563e+07
PRODUCTIVITY           1.678733e+07
GAME                   1.583757e+07
TRAVEL_AND_LOCAL       1.398408e+07
TOOLS                  1.069525e+07
NEWS_AND_MAGAZINES     9.549178e+06
BOOKS_AND_REFERENCE    8.767812e+06
SHOPPING               7.036877e+06
PERSONALIZATION        5.201483e+06
WEATHER                5.074486e+06
HEALTH_AND_FITNESS     4.188822e+06
MAPS_AND_NAVIGATION    4.056942e+06
SPORTS                 3.638640e+06
EDUCATION              3.082018e+06
FAMILY                 2.691618e+06
FOOD_AND_DRINK         1.924898e+06
ART_AND_DESIGN         1.905352e+06
BUSINESS               1.712290e+06
LIFESTYLE              1.437816e+06
FINANCE                1.387692e+06
HOUSE_AND_HOME         1.313682e+06
DATING                 8.540288e+05
COMICS                 8.176573e+05
AUTO_AND_VEHICLES      6.473

According to this niche, communication applications have the highest number of installation, following by entertainment, tools, social, photography and video players. Let's see what are these apps in communcation category.

---
Theo phân khúc này, các ứng dụng truyền thông có số lượng cài đặt cao nhất, tiếp theo là giải trí, công cụ, mạng xã hội, nhiếp ảnh và trình phát video. Hãy xem những ứng dụng này trong danh mục truyền thông là gì.


In [221]:
free_appandr_df.loc[free_appandr_df['Category']=='COMMUNICATION', ['App', 'Installs']].sort_values(by='Installs')

App      Installs
4921                                  BA SALES  1.000000e+00
5376                    Of the wall Arapaho bk  5.000000e+00
8964                        Test Server SMS FA  5.000000e+00
7051                        Hlášenírozhlasu.cz  1.000000e+01
4143                 Oklahoma Ag Co-op Council  1.000000e+01
...                                        ...           ...
304                                      Gmail  1.000000e+09
349              Skype - free IM & video calls  1.000000e+09
302               Google Chrome: Fast & Secure  1.000000e+09
300                         WhatsApp Messenger  1.000000e+09
299   Messenger – Text and Video Chat for Free  1.000000e+09

[287 rows x 2 columns]

This again shows that there are some giants dominating the whole category, which could lead to possibly false direction of analysis.

Let's look at book category and see if the market is potential or not.


---
Điều này một lần nữa cho thấy rằng có một số gã khổng lồ đang thống trị toàn bộ danh mục, điều này có thể dẫn đến hướng phân tích sai lầm.

Chúng ta hãy nhìn vào danh mục sách và xem thị trường có tiềm năng hay không.


In [220]:
free_appandr_df.loc[free_appandr_df['Category']=='BOOKS_AND_REFERENCE', ['App', 'Installs']].sort_values(by='Installs')

App      Installs
8794                   EW PDF  5.000000e+00
7050                  CZ-Help  5.000000e+00
3632  Guide for R Programming  5.000000e+00
7025   BibleRead En Cy Zh Yue  5.000000e+00
3635      Learn R Programming  1.000000e+01
...                       ...           ...
3131                    Bible  1.000000e+08
4661  Audiobooks from Audible  1.000000e+08
139      Wattpad 📖 Free Books  1.000000e+08
143             Amazon Kindle  1.000000e+08
152         Google Play Books  1.000000e+09

[190 rows x 2 columns]

The results above show that software for processing and reading ebooks, as well as various collections of libraries and dictionaries are the dominants. So it will not be a good idea to builid simila apps as there will be quite significant competition.

There are some apps build around Al-Quran, which means that building an app around a popular book can be profitable for both markets. However, there must be some different features inside the application to make it outstanding from other applications, especially libraries. Some features could be audiobook, daily quotes from book, quizzes and discussion where people can discuss the book.

---

Kết quả trên cho thấy phần mềm xử lý và đọc sách điện tử cũng như các bộ sưu tập thư viện và từ điển khác nhau đang chiếm ưu thế. Vì vậy, việc xây dựng ứng dụng simila sẽ không phải là một ý tưởng hay vì sẽ có sự cạnh tranh khá lớn. Có một số ứng dụng được xây dựng xoay quanh Al-Qur'an, điều đó có nghĩa là việc xây dựng một ứng dụng xoay quanh một cuốn sách nổi tiếng có thể mang lại lợi nhuận cho cả hai thị trường. Tuy nhiên, bên trong ứng dụng phải có một số tính năng khác biệt để làm cho nó nổi bật so với các ứng dụng khác, đặc biệt là các thư viện. Một số tính năng có thể là sách nói, trích dẫn hàng ngày từ sách, câu đố và thảo luận nơi mọi người có thể thảo luận về cuốn sách.

## Conclusions

I have analyzed the web scrapped data sets of App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for these markets.

I found out that the book genre application is quite potential with very few popular applications compared to other popular genres such as communication or photography. That taking a recently popular book and turning it into an app could be profitable for App Store and Google Play markets. Since the book app market is now full of libraries, the app should contain special features to make it outstanding in the market. Such features could be audio version, daily quotes and quizzes, a discussion forum or review section.

---

Tôi đã phân tích bộ dữ liệu bị loại bỏ trên web của ứng dụng di động App Store và Google Play với mục tiêu đề xuất hồ sơ ứng dụng có thể mang lại lợi nhuận cho các thị trường này. Tôi nhận thấy ứng dụng thể loại sách khá tiềm năng với rất ít ứng dụng phổ biến so với các thể loại phổ biến khác như giao tiếp hay nhiếp ảnh. Việc lấy một cuốn sách phổ biến gần đây và biến nó thành một ứng dụng có thể mang lại lợi nhuận cho thị trường App Store và Google Play. Vì thị trường ứng dụng sách hiện có rất nhiều thư viện nên ứng dụng phải chứa các tính năng đặc biệt để làm cho ứng dụng trở nên nổi bật trên thị trường. Các tính năng như vậy có thể là phiên bản âm thanh, trích dẫn và câu đố hàng ngày, diễn đàn thảo luận hoặc phần đánh giá.